astropy/coordinates/representation.py

```diff
@@ -221,6 +228,143 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def __eq__(self, other):
+        """Check for equality with another representation.
+
+        If either ``self`` or ``other`` are not scalar, this returns a boolean
+        array, just like is the case for ``ndarray``.  Elements are `True` only
+        if all components of that element are exactly equal.
+        """
+        try:
+            # this ensures that other can be compared sensibly to self.
+            # note that it immediately returns if the classes are the same.
+            other = other.represent_as(self.__class__)
+        except Exception:
+            return False
+
+        # Ensure we return a bool or array of bool as appropriate.
+        return functools.reduce(np.logical_and,
+                                (getattr(self, component) ==
+                                 getattr(other, component)
+                                 for component in self.components))
+
+    def __ne__(self, other):
+        return np.logical_not(self.__eq__(other))
+
+    def _scale_operation(self, op, other):
+        results = []
+        for component, cls in self.attr_classes.items():
+            value = getattr(self, component)
+            if issubclass(cls, Angle):
+                results.append(value)
+            else:
+                result = op(value, other)
```

Comment:
Any representation can be multiplied by any quantity: it is just the ones that carry dimensional information (i.e., not the angles, so, e.g., r and z for cylindrical) that get multiplied.